### PR TITLE
enable auth for prado

### PR DESCRIPTION
**Pull request by alfredodeza**
Implements simple HTTP auth for prado

### CODE REVIEW

File: deploy/playbooks/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -26,6 +27,23 @@
   pip: name='git+https://github.com/alfredodeza/{{ app_name }}#egg={{ app_name }}' virtualenv={{ app_home }}
   changed_when: False
 
+# this will only be used if the credentials where not passed in explicitly over
+# the command line as --extra-vars and the app_home directory was not created
+# (as in a new deployment). Once the app is deployed once, the api/pass should
+# stay consistent
+- name: generate pseudo-random password for basic http auth
```
Comment (andrewschoen) [inline deploy/playbooks/roles/common/tasks/main.yml:34]: Wouldn't this change the password every time?

File: deploy/playbooks/roles/common/tasks/main.yml [hunk 2]
```diff
@@ -26,6 +27,23 @@
...
+  shell: python -c "exec 'import os; print os.urandom(30).encode(\'base64\')[:${length}]'"
+  register: api_password
+  changed_when: false
+
+- name: create the prod api credentials file
+  template:
+    src: prod_api_creds.py.j2
+    dest: "{{ app_home }}/src/{{ app_name }}/prod_api_creds.py"
+  when: (api_password is defined or api_user is defined) or
```
Comment (andrewschoen) [inline deploy/playbooks/roles/common/tasks/main.yml]: api_password is always defined, right? So wouldn't this generate a new password on every deploy?

Comment (alfredodeza) [inline deploy/playbooks/roles/common/tasks/main.yml]: dammit this should be api_key not api_password

File: deploy/playbooks/roles/common/templates/prod_api_creds.py.j2 [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Basic HTTP Auth credentials
+api_user = "{{ api_user | default('admin') }}"
+api_key = "{{ api_key | default('secret') }}"
```
Comment (andrewschoen) [inline deploy/playbooks/roles/common/templates/prod_api_creds.py.j2]: don't you want the default to be `api_key_generated` here?

**Pull request merged**